src/components/tag-box/tag-box.js

```diff
@@ -0,0 +1,367 @@
+    if (this.canOpenInput()) {
+      if (
+        event.relatedTarget === this.containerElement ||
+        this.containerElement.contains(event.relatedTarget)
+      )
+        return;
+      this.containerElement.removeChild(this.inputElement);
+    }
+  }
+
+  handleInputTextKeyUp(event) {
+    if (this.canOpenInput()) {
+      if (event.key === 'Enter' && event.target.value) {
+        const newTagBox = createTagBoxItem(
+          event.target.value
+        ).content.cloneNode(true);
+        this.containerElement.appendChild(newTagBox);
+        this.containerElement.lastElementChild.addEventListener(
+          'onRemoveItem',
+          this.handleTagBoxItemRemoveClick.bind(this)
+        );
+        this.inputElement.blur();
+        this.inputElement.value = '';
+        this.dispatchEvent(
+          new CustomEvent('onAddItem', {
+            detail: { item: this.containerElement.lastElementChild },
+          })
+        );
+        this.dispatchChangeEvent();
+      }
+    }
+  }
+
+  handleTagBoxItemRemoveClick(event) {
+    const isInHidden = this.hiddenItemElements.includes(event.target);
+    this.containerElement.removeChild(event.target);
+    this.dispatchChangeEvent();
+    if (this.type === 'message-box') return;
+
+    if (
+      this.hiddenNumberElement &&
+      this.containerElement.contains(this.hiddenNumberElement)
+    ) {
+      this.containerElement.removeChild(this.hiddenNumberElement);
+    }
+
+    if (isInHidden) {
+      this.hiddenItemElements = this.hiddenItemElements.filter(
+        (item) => item !== event.target
+      );
+      if (this.hiddenItemElements.length > 0) {
+        this.hiddenNumberElement.innerText = `+${this.hiddenItemElements.length}`;
+        this.containerElement.appendChild(this.hiddenNumberElement);
+      }
+    } else {
+      if (this.hiddenItemElements.length) {
+        const hiddenItems = [...this.hiddenItemElements].reverse();
+        const removeIndex = [];
+        let isStop = false;
+        this.hiddenItemElements = hiddenItems
+          .filter((hidItem, index) => {
+            if (!isStop) {
+              const itemWidth = this.realWidth(hidItem);
+              const containerWidth = this.realWidth(
+                this.containerElement,
+                true
+              );
+              if (
+                containerWidth + itemWidth + lastItemWidth <=
+                this.containerElement.offsetWidth
+              ) {
+                this.containerElement.appendChild(hidItem);
+                removeIndex.push(index);
+                return false;
+              }
+            }
+            isStop = true;
+            return true;
+          })
+          .reverse();
+        if (this.hiddenItemElements.length > 0) {
+          this.hiddenNumberElement.innerText = `+${this.hiddenItemElements.length}`;
+          this.containerElement.appendChild(this.hiddenNumberElement);
+        }
+      }
+    }
+  }
+
+  canOpenInput() {
```

Comment:
convert to 

get canInput () {

return this.show || this.type !== 'text-box';

}